api/layer/object_cache.go

```diff
@@ -0,0 +1,86 @@
+package layer
+
+import (
+	"sync"
+	"time"
+
+	cid "github.com/nspcc-dev/neofs-api-go/pkg/container/id"
+)
+
+/*
+	This is an implementation of a cache for ListObjectsV2 which we return to users by ContinuationToken.
+
+	The cache is a map which has a key: (access_key from AccessBox) + container_id and a value: list of objects with
+	creation time. After putting a record we start a timer (via time.AfterFunc) that removes the record after
+	defaultCacheLifetime value.
+
+	ContinuationToken in our gateway is an objectID in NeoFS.
+
+	We don't keep ContinuationToken in this structure because we assume that users who received the token can reconnect
+	to other gateways and they should be able to get a list of objects.
+	When we receive the token from the user we just try to find the cache and then we return the list of objects which
+	starts from this token (i.e. objectID).
+*/
+
+// ObjectsListV2Cache provides interface for cache of ListObjectsV2 in a layer struct.
+type (
+	ObjectsListV2Cache interface {
+		GetCache(token string, key string) []*ObjectInfo
+		PutCache(key string, objects []*ObjectInfo)
+	}
+)
+
+var (
+	defaultCacheLifetime = time.Second * 60
+)
+
+type (
+	listObjectsCache struct {
+		caches map[string]cache
+		mtx    sync.RWMutex
+	}
+	cache struct {
+		list    []*ObjectInfo
+		created time.Time
```

Comment:
This field is not used (we only set it).

Reply:
I guess with `time.AfterFunc()` we can just

```go
listObjectsCache struct {
	caches map[string][]*ObjectInfo
	mtx    sync.RWMutex
}
```

Reply:
Maybe we can leave cache struct, doesn't it make code more readable? And this struct make more clear what does it mean `cache` in code.